### DISCUSSION TITLE
Using Environment Variables with period in the name - Not passed to build processes

### DISCUSSION THREAD

**Discussion by simplemes-old** [community/community#25573] [Actions]
<p>All,</p>
<p>An odd behavior.  I need to set an environment variable with a period in the name (eframe.key).  The action YAML file is:</p>
<pre><code>name: BuildAndTestAll

on: [push]

jobs:
  build:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v1
      - name: Set up JDK 1.8
        uses: actions/setup-java@v1
        with:
          java-version: 1.8
      - name: Build and Test
        env:
          # A key to use for testing, not used in production.
          eframe.key: someKey890123456789012345678901234567890
          eframeKey: othersomeKey890123456789012345678901234567890
        run: ./gradlew cleanAll testAll -DwriteTests=false
</code></pre>
<p>The variable eframeKey works fine and is in the environment.  The eframe.key is not.  I am using Micronaut and it seems to need the variable with a period in the name (not sure why yet).</p>
<p>I have  a short-term work around using a default value in the code, but I need to avoid that for security reasons.</p>
<p>Is period just a reserved character for the names?</p>
<p>The repository is: <a href="https://github.com/simplemes/simplemes-core" rel="nofollow noopener">https://github.com/simplemes/simplemes-core</a></p>

**Comment by SH4FI-DEV** [comment]
<p>Did you tried using underscore instead of period?</p>

**Comment by Yanjingzhu** [comment]
<p>Hi simplemes,</p>
<p>As  <a href="/t5/user/viewprofilepage/user-id/56065">shafi1996</a> suggested, when name an environment variable, please use '_ ’ instead of  '. ’ . </p>
<p>Use <strong>eframe_key</strong> instead of e <strong>frame.key.</strong></p>
<p>There is the document for Using environment variables: <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/using-environment-variables" rel="nofollow noopener">https://help.github.com/en/actions/automating-your-workflow-with-github-actions/using-environment-variables</a> </p>

**Comment by simplemes-old** [comment, marked as answer]
<p>Thanks for the feedback.   I did try using underscore, but I found a problem with Gradle not passing the env variable to the forked test process.  It only seems to pass variables with a period in them :(.</p>
<p>I found a work-around with Micronaut using the application-test.yml configuration file.  That file provides a value without the need for an environment variable.  I will mark this as the answered, just to help other micronaut users.</p>
<p>BTW, the link to the doc did not work.  I used: <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/using-environment-variables" rel="nofollow noopener">https://help.github.com/en/actions/automating-your-workflow-with-github-actions/using-environment-variables</a> </p>
<p>I could not find anyting saying period (decimal) is not allowed.  It is implied that there is some structure to the variable naming scheme.  The  syntax: <a href="https://help.github.com/en/github/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_idenv" rel="nofollow noopener">jobs.job.env</a> gave me a clue that period might not be supported.</p>
<p>Not a big problem, but just for future searchers…</p>
